astropy/modeling/tests/test_quantities_parameters.py

```diff
@@ -0,0 +1,476 @@
+    """
+    Basic test of comparison operations on properties with units.
+    """
+
+    g = Gaussian1D(1 * u.J, 1 * u.m, 0.1 * u.m)
+
+    # Essentially here we are checking that parameters behave like Quantity
+
+    assert g.mean == 1 * u.m
+    assert 1 * u.m == g.mean
+    assert g.mean != 1
+    assert 1 != g.mean
+
+    assert g.mean < 2 * u.m
+    assert 2 * u.m > g.mean
+
+    with pytest.raises(UnitsError) as exc:
+        g.mean < 2
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+    with pytest.raises(UnitsError) as exc:
+        2 > g.mean
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+    g = Gaussian1D([1, 2] * u.J, [1, 2] * u.m, [0.1, 0.2] * u.m)
+
+    assert g.mean == [1, 2] * u.m
+    assert np.all([1, 2] * u.m == g.mean)
+    assert g.mean != [1, 2]
+    assert np.all([1, 2] != g.mean)
+
+    with pytest.raises(UnitsError) as exc:
+        g.mean < [3, 4]
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+    with pytest.raises(UnitsError) as exc:
+        [3, 4] > g.mean
+    assert exc.value.args[0] == ("Can only apply 'less' function to "
+                                 "dimensionless quantities when other argument "
+                                 "is not a quantity (unless the latter is all "
+                                 "zero/infinity/nan)")
+
+
+def test_parameter_unit_linking_gaussian():
+    """
+    For some models, units of parameters should be related/consistent. We check
+    the case of the Gaussian here, where the mean and stddev should have the
+    same units.
+    """
+
+    with pytest.raises(UnitsError) as exc:
+        g = Gaussian1D(1 * u.J, 1. * u.m, 0.1 * u.s)
+    assert exc.value.args[0] == ("Parameters 'mean' and 'stddev' should "
+                                 "have matching units")
+
+
+@pytest.mark.xfail
+def test_parameter_unit_linking_custom():
+
+    # TODO: this is just an idea for an API, and open to feedback on the best
+    # way to do this. The current approach is to provide a list of tuples,
+    # where each tuple is a group of units that should have linked units. I
+    # found it easier conceptually to keep this outside the parameter
+    # definition, otherwise we have to navigate all parameters and construct a
+    # tree.
+
```

Comment:
What of the model is not pure numpy code? E.g. if I have a model that uses some scipy function all the units of the parameters will be lost. There is no way to find out if the initial values where compatible or not.
(I recently had to make a model that required a call to "odesolve". I know analytically what the relation between the parameters is and which of them need compatible units, but evaluation the model will always work, because odesolve will only see the values, not the units.)